### PR TITLE
ZBUG-1096: fixing IMAP synchronization problem on Mobile App

### PR DESCRIPTION
**Pull request by dasiyogesh**
**Issue:**
Shared folders in IMAP synchronization are getting wrongly mapped to owners folders and shared ones.

**Fix:**
Separated the list of owner & shared folders and passed the list.
For shared folder list removed the folder mapping.

**Testing Done:**
Shared root folder of account with sharee.
Tested the folders are mapped properly and shared folders show content within them and not the owners ones.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/imap/ImapHandler.java [hunk 1]
```diff
@@ -2437,43 +2476,45 @@ private String getFolderAttrs(ImapPath path, byte returnOptions, Map<ImapPath, I
...
-                    if (query != null) {
-                        if (query.equalsIgnoreCase("is:flagged")) {
-                            attrs.append(attrs.length() == 0 ? "" : " ").append(returnXList ? "\\Starred" : "\\Flagged");
-                        } else if (query.equalsIgnoreCase("is:local")) {
-                            attrs.append(attrs.length() == 0 ? "" : " ").append(returnXList ? "\\AllMail" : "\\All");
+        // don't add folder mapping for mounted folders
+        if (!isMounted) {
+            //partial support for special use RFC 6154; return known folder attrs
+            //we also keep support for non-standard XLIST attributes for legacy clients that may still use them
+            if ((DebugConfig.imapForceSpecialUse || (returnOptions & RETURN_XLIST) != 0) && path.belongsTo(credentials)) {
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/imap/ImapHandler.java]: why not add isMounted clause to this if
if ((DebugConfig.imapForceSpecialUse || (returnOptions & RETURN_XLIST) != 0) && path.belongsTo(credentials) && !isMounted)

File: store/src/java/com/zimbra/cs/imap/ImapHandler.java [hunk 2]
```diff
@@ -2295,6 +2269,69 @@ private boolean doLIST(String tag, String referenceName, Set<String> mailboxName
         return true;
     }
 
+    private void accumulateSelectedFolders(Map<ImapPath, ItemId> paths, Set<ImapPath> selected, Set<String> remoteSubscriptions, String owner) throws ServiceException {
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/imap/ImapHandler.java]: Can we add unit test cases for these methods

File: store/src/java/com/zimbra/cs/imap/ImapHandler.java (hunk 1 above)
Comment (dasiyogesh) [reply store/src/java/com/zimbra/cs/imap/ImapHandler.java]: Included mounted folder check within in if block

File: store/src/java/com/zimbra/cs/imap/ImapHandler.java (hunk 2 above)
Comment (dasiyogesh) [reply store/src/java/com/zimbra/cs/imap/ImapHandler.java]: will check for it

**Pull request merged**